### DISCUSSION TITLE
Support Static File Routes to have ISR

### DISCUSSION THREAD

**Discussion by ovflowd** [vercel/next.js#48909] [Ideas]
### Goals

1. Faster builds by supporting ISR on full-static Websites or Websites that are mostly Static
2. Faster builds by supporting ISR on routes that the maintainers of a specific App deem less essential to be built on the cold/warm initial build.
3. Fine-tuned control over what routes should be omitted on build (i.e. on `getStaticProps`)
  - No need for revalidation/interval as content is static
5. Fine-tuned control over what routes should be omitted on build (i.e. via Turborepo with some extra property to configure the routes with patterns/blobs and still respecting changes on `inputs`)

### Background

1. You might have hundreds/thousands of static files. For example, blog posts or your App is localised in multiple languages.
2. You don't need the blog posts to be part of your website's initial cold (or even warm) build. These can be built on-demand via ISR as your user hits these endpoints.
3. This would allow faster builds on Vercel or Next.js in general, if you want to avoid having a full-build (e.g. staging environments, local development full-build)
4. You still want to check that changes are built, usually in relation to existing content. (E.g. CI pipelines, you want to build only the changes in this case...)
5. It Feels like something that could be integrated within Turborepo (?)

There are currently no alternatives, and I bet other people want this feature too! As it can be pretty damn helpful and reduce cold builds significantly.

### Proposal

This could be implemented differently and supported via Vercel Build Outputs, `getStaticProps`, and Turborepo. Allowing more control over ISR builds on non-Next.js dependencies and a pure Next.js approach would be ideal.

I'm interested in contributing and also in being early adopter (OpenJS Foundation, Node.js)

**Comment by ovflowd** [comment]
cc @Ethan-Arrowood @shuding @leerob for visibility (As I think some of you are my key-connections at Vercel 🙈)

**Reply by leerob** [reply]
You can have ISR semantics for static files with Route Handlers: https://beta.nextjs.org/docs/routing/route-handlers

**Reply by ovflowd** [reply]
That seems to be an App Router-only feature?

**Reply by leerob** [reply]
Correct, but it's entirely incrementally adoptable. Meaning you could keep `pages/` exactly the same, but have `app/my-file.txt/route.ts` living in the same codebase.

**Reply by ovflowd** [reply]
So I could have `/pages/en/blog/[some-static-file-in-file-system]` and then have `app/en/blog/route.ts`?

**Reply by leerob** [reply]
As long as there isn't a conflicting route 👍

**Reply by ovflowd** [reply]
Gotcha, not sure I got this 100% but will play around :)

Appreciate the support!
